File: problem7.py
# write a function to find the 10001st prime number
def prime(x):
    for i in range(2, x):
        if x % i == 0:
            return False
    return True

def prime10001():
    count = 0
    for i in range(2, 1000000000):
        if prime(i):
            count += 1
            if count == 10001:
                return i
            
print(prime10001())

File: problem7.js
// check if a number is prime
const prime = (x) => {
    for (let i = 2; i <= Math.sqrt(x); i++) {
        if (x % i === 0) return false;
    }
    return true;
} 

// counter variable
let count = 0;

// loop through numbers until we find the 10001st prime
let i = 2;

while(count < 10002) {
    if(prime(i)) {
        count++;
        console.log(count, i);
    }
    i++;
}

//console.log(i - 1);